internal/contractregistry/contractstore.go

```diff
@@ -161,22 +175,42 @@ func (cs *contractStore) GetContractByAddress(addrHex string) (*ContractInfo, er
 	return info.(*ContractInfo), nil
 }
 
-func (cs *contractStore) GetABIByID(id string) (*messages.DeployContract, *ABIInfo, error) {
+func (cs *contractStore) GetABI(location *ABILocation, refresh bool) (*messages.DeployContract, string, error) {
+	switch location.ABIType {
```

Comment:
Is there any better way to generically store a reference to an ABI? I could only pare down to these 3 different ways of retrieving an ABI, so I was left with this "ABILocation" that pairs an enum with a string identifier...